blockchain/My-Wallet-V3-iOS
# feat(KYC): show state selection screen if selected country has states

## Objective

Display state selection screen if the selected country during KYC has states.

## Description

Updated `KYCCountry` to decode the `states` property so that if the selected country in the `KYCCountrySelectionController` has states, `KYCStateSelectionController` will be presented. I could've probably reused `KYCCountrySelectionController` to display the list of states, but figured it's better to create a new view controller for this for better separation.

Other changes:
* Genericized `CountriesMap` to `SearchableMap` 
* Created new and updated existing unit tests

## How to Test

Go through KYC, select "United States", and verify that you are brought to the state selection screen.

## Screenshot/Design assessment

![simulator screen shot - iphone 5s - 2018-10-09 at 15 05 35](https://user-images.githubusercontent.com/38220701/46702402-d2369600-cbd7-11e8-8335-cbb5b18b7002.png)

![simulator screen shot - iphone 5s - 2018-10-09 at 15 28 23](https://user-images.githubusercontent.com/38220701/46702453-fb572680-cbd7-11e8-844f-c8601ee9f7e1.png)

## Merge Checklist
